dev-python/requests-cache/requests-cache-0.5.2.ebuild

```diff
@@ -0,0 +1,77 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+PYTHON_REQ_USE="sqlite"
+
+inherit distutils-r1
+
+HOMEPAGE="https://pypi.org/project/requests-cache/"
+DESCRIPTION="Persistent cache for requests library"
+SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
+
+RDEPEND="dev-python/requests[${PYTHON_USEDEP}]"
+DEPEND="doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )"
+
+distutils_enable_tests unittest
+
+src_prepare() {
+	# Ships with incorrect mock import
+	sed -e 's/import mock/import unittest.mock as mock/' -i tests/test_cache.py || die
+
+	# Slew of tests that violate network-sandbox
+	sed -e 's/test_requests_from_cache/_test_requests_from_cache/' \
+		-i tests/test_monkey_patch.py || die
+	sed -e 's/test_attr_from_cache_in_hook/_test_attr_from_cache_in_hook/' \
+		-e 's/test_cache_unpickle_errors/_test_cache_unpickle_errors/' \
+		-e 's/test_close_response/_test_close_response/' \
+		-e 's/test_content_and_cookies/_test_content_and_cookies/' \
+		-e 's/test_delete_urls/_test_delete_urls/' \
+		-e 's/test_disabled/_test_disabled/' \
+		-e 's/test_enabled/_test_enabled/' \
+		-e 's/test_expire_cache/_test_expire_cache/' \
+		-e 's/test_from_cache_attribute/_test_from_cache_attribute/' \
+		-e 's/test_get_parameters_normalization/_test_get_parameters_normalization/' \
+		-e 's/test_get_params_as_argument/_test_get_params_as_argument/' \
+		-e 's/test_gzip_response/_test_gzip_response/' \
+		-e 's/test_headers_in_get_query/_test_headers_in_get_query/' \
+		-e 's/test_hooks/_test_hooks/' \
+		-e 's/test_https_support/_test_https_support/' \
+		-e 's/test_ignore_parameters_get/_test_ignore_parameters_get/' \
+		-e 's/test_ignore_parameters_post/_test_ignore_parameters_post/' \
+		-e 's/test_ignore_parameters_post_json/_test_ignore_parameters_post_json/' \
+		-e 's/test_ignore_parameters_post_raw/_test_ignore_parameters_post_raw/' \
+		-e 's/test_post/_test_post/' \
+		-e 's/test_post_data/_test_post_data/' \
+		-e 's/test_post_parameters_normalization/_test_post_parameters_normalization/' \
+		-e 's/test_post_params/_test_post_params/' \
+		-e 's/test_remove_expired_entries/_test_remove_expired_entries/' \
+		-e 's/test_response_history/_test_response_history/' \
+		-e 's/test_response_history_simple/_test_response_history_simple/' \
+		-e 's/test_return_old_data_on_error/_test_return_old_data_on_error/' \
+		-e 's/test_stream_requests_support/_test_stream_requests_support/' \
+		-i tests/test_cache.py || die
+
+	# This throws many errors with network-sandbox, but doesn't fail the build
+	sed -e 's/test_caching_with_threads/_test_caching_with_threads/' \
+		-i 'tests/test_thread_safety.py' || die
+
+	default
+}
+
+# Testsuite excels in tests connecting to the network via local server daemons
+python_compile_all() {
+	use doc && emake -C docs html
+}
```

Comment:
try killing `python_compile_all` and `python_install_all` and just doing `distutils_enable_sphinx docs` after `distutils_enable_tests` (and remove sphinx from `DEPEND`, it belongs in `BDEPEND` anyhow)

Reply:
Sphinx dependency dropped, python* functions obsoleted by distutils_enable_sphinx and dropped. Added 3.9 while at it too (tests pass for me, dev-python/requests supports it).